sympy/core/containers.py

```diff
@@ -269,6 +269,14 @@ def _sorted_args(self):
         return tuple(sorted(self.args, key=default_sort_key))
 
 
+converter[dict] = lambda d: Dict(*d.items())
+# a user can pass defaultdict to Dict but the
+# result is dict-like not defaultdict-like, so
+# by setting the converter to None we will
+# catch this case and sympify will raise an error
+converter[defaultdict] = None
+converter[OrderedDict] = None  # the same applies for OrderedDict
```

Comment:
I guess I can understand disabling defaultdict (though honestly I would just let it work), but OrderedDict is the same as dict, except ordered. And sympy.Dict is immutable and ordered, so I think it's fine.

Reply:
> I would just let it work

all dict-types are handled the same way, now